app/controllers/regions_controller.rb

```diff
@@ -1,20 +1,47 @@
 class RegionsController < ApplicationController
+
+  before_filter :correct_manager, except: [:index, :show]
+
   # before_action :authenticate_user!
   def index
     @regions = Region.all
-    @cities = Cities.all
+    @cities = City.all
   end
 
   def show
     @region = Region.find(params[:id])
     @guest = session[:guest_id] ? Guest.find(session[:guest_id]) : Guest.new
+    @cities = City.where(region_id: params[:id])
   end
 
   def new
   	@region = Region.new
   end
 
   def create
-  	Region.create(params[:region])
+    @region = Region.new(params[:region])
+    @is_created=true
+    if @region.valid?
+      @region.save
+    else
+      @is_created=false
+    end
+
+  end
+
+  def add_city
+    @is_valid = true
+    if params[:city][:id] != nil && params[:city][:id] != ""
```

Comment:
be robust. change to
`if params[:city].present? && params[:city][:id].present?`
`.present?` excludes nil and empty strings